syntax = "proto3";

package envoy.config.trace.v3;

import "google/protobuf/duration.proto";

import "udpa/annotations/migrate.proto";
import "udpa/annotations/status.proto";
import "udpa/annotations/versioning.proto";
import "validate/validate.proto";

option java_package = "io.envoyproxy.envoy.config.trace.v3";
option java_outer_classname = "DatadogProto";
option java_multiple_files = true;
option go_package = "github.com/envoyproxy/go-control-plane/envoy/config/trace/v3;tracev3";
option (udpa.annotations.file_migrate).move_to_package = "envoy.extensions.tracers.datadog.v4alpha";
option (udpa.annotations.file_status).package_version_status = ACTIVE;

// [#protodoc-title: Datadog tracer]

// Remote Configuration
message DatadogRemoteConfig {
  // Allows to configure and change the tracer behaviour from Datadog's UI.
  // Default value is `false`, which disable the feature.
  // [#comment: This feature can drastically increase the number of connections to the Datadog Agent.
  // This is the consequence of each tracer regularly polling for configuration updates, and the number
  // of tracer is the product of the number of listeners and worker threads.]
  bool enabled = 1;

  // Frequency at which new configuration updates are queried.
  google.protobuf.Duration polling_interval = 2;
}

// Configuration for the Datadog tracer.
// [#extension: envoy.tracers.datadog]
message DatadogConfig {
  option (udpa.annotations.versioning).previous_message_type =
      "envoy.config.trace.v2.DatadogConfig";

  // The cluster to use for submitting traces to the Datadog agent.
  string collector_cluster = 1 [(validate.rules).string = {min_len: 1}];

  // The name used for the service when traces are generated by envoy.
  string service_name = 2 [(validate.rules).string = {min_len: 1}];

  // Optional hostname to use when sending spans to the collector_cluster. Useful for collectors
  // that require a specific hostname. Defaults to :ref:`collector_cluster <envoy_v3_api_field_config.trace.v3.DatadogConfig.collector_cluster>` above.
  string collector_hostname = 3;

  // Remote Configuration.
  DatadogRemoteConfig remote_config = 4;
}
